### PR TITLE
Branch auto switcher

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
Introduces a GitHub Action which switches the branches automatically, basically automating both preparing for releases like https://github.com/gitpod-io/openvscode-releases/pull/21, but also back to using `main`, like https://github.com/gitpod-io/openvscode-releases/pull/22.

### CODE REVIEW

**Comment by mustard-mh** [comment]
We need to update node_modules' cache sometimes if packages changes, this can be an option of action args too

**Comment by jeanp413** [comment]
Lets use `actions/github-script@v6` as, you can take a look at [new-upstream-release.yml](https://github.com/gitpod-io/openvscode-releases/blob/main/.github/workflows/new-upstream-release.yml) action

**Comment by filiptronicek** [comment]
Agree that's the way to do this, I just wasn't very familiar with `actions/github-script@v6` (didn't know it could use NPM packages other than GitHub's octokit). I pushed the version utilizing it now.

File: .github/workflows/switch.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          path: '/tmp/gitpod/'
+          repository: 'gitpod-io/gitpod'
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Install dependencies
+        run: npm i
```
Comment (jeanp413) [inline .github/workflows/switch.yml:25]: Does this work? Since we don't have a package.json I think we need to install dependencies explicitly

Comment (filiptronicek) [reply .github/workflows/switch.yml:25]: 🤦😆 good catch!

File: .github/workflows/switch.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      - name: Run the branch switch-a-roo
+        uses: actions/github-script@v6
+        with:
+          script: |
+            import semver from 'semver';
```
Comment (jeanp413) [inline .github/workflows/switch.yml:30]: Same here, I have not tested it before but I think you need to use `require` instead of import

Comment (filiptronicek) [reply .github/workflows/switch.yml:28]: Looks like you're correct, thank you! https://github.com/actions/github-script#use-esm-import

**Comment by jeanp413** [comment]
Let's do this:
- For pointing to the new branch:
  - call this action if `notify-new-release` succeeds, you'll need change it so it returns the new branch name and then pass it as an parameter to this script,
  - then this action should create the new branches in openvscodeserver based on the `main` / `gp-code/main` branches
  - finally just commit the changes updating to the new branch in this repo no need for PR(bonus points create a slack notification to with the commit URL ), for gp-code repo keep the PR as it will require someone to approve it.

- For pointing back to main check:
  - check for the new release in vscode repo and do the same as the last step in `pointing to the new branch`

**Comment by filiptronicek** [comment]
@jeanp413 your suggestions sound good (like not creating a PR for this repo). I'll adjust the Action based on them.

> you'll need change it so it returns the new branch name and then pass it as a parameter to this script

I don't think it's necessary to make any modifications here. The Action already queries the next minor version release.

> bonus points create a slack notification to with the commit URL

Woooo! Bonus points!!!!

File: .github/workflows/switch.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+name: Switch workflow branches
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to check"
+        required: false
+    workflow_run: 
+      workflows: [Notify new upstream release]
+      types: [completed]
```
Comment (jeanp413) [inline .github/workflows/switch.yml:10]: `Notify new upstream release` will always succeed and it runs every day, this action should only run when a new upstream branch is created, I was thinking calling this action [like this](https://github.com/gitpod-io/openvscode-releases/blob/41232a840ca81f4101c7738ebd645cd97683df0d/.github/workflows/release.yml#L272-L282)

Comment (filiptronicek) [reply .github/workflows/switch.yml]: thanks @jeanp413, changed it to work like that in [`94a4531` (#23)](https://github.com/gitpod-io/openvscode-releases/pull/23/commits/94a4531de183e1a8695849c4b2ca6ee046da8141).

**Comment by jeanp413** [comment]
Let's try this

**Review by jeanp413 — APPROVED** [review]

**Comment by jeanp413** [comment]
Well it doesn't work xD @filiptronicek try to fix it when you have time

**Comment by filiptronicek** [comment]
Aaah it seems `semver` doesn't like `1.70` as an input, so it must be `1.70.0` 👀. There may be more bugs, but this is one that probably was the reason the workflow failed, @jeanp413.

**Pull request merged**